components/sup/src/manager/service/config.rs

```diff
@@ -316,41 +316,39 @@ struct Cfg {
     environment: Option<toml::Value>,
 }
 
-// Shallow merges two toml tables.
-fn toml_merge(existing: &toml::Table, ovr: &toml::Table) -> toml::Table {
-    let mut final_map = existing.clone();
-    for (ovr_key, ovr_value) in ovr.iter() {
-
-        // attempt to merge nested tables
-        let existing_v = existing.get(ovr_key).and_then(|v| v.as_table());
-        let ovr_v = ovr.get(ovr_key).and_then(|v| v.as_table());
-        if let None = existing.get(ovr_key) {
-            final_map.insert(ovr_key.clone(), ovr_value.clone());
+// Recursively merges two TOML tables
+fn toml_merge(me: &toml::Table, other: &toml::Table) -> toml::Table {
+    let mut result = me.clone();
```

Comment:
Since this is a generic function for performing TOML merges (albeit private to our Config class) I would be careful about weighing on how small our current data set is. Somebody is going to call this when one of us isn't paying attention on a struct that could dynamically grow to massive sizes.

At the very least I would rename this to something like `toml_merge_and_new_from()` or something that indicates it's evil allocation tendencies :)